x-pack/plugins/ml/public/application/components/data_grid/use_data_grid.tsx

```diff
@@ -131,6 +134,16 @@ export const useDataGrid = (
     });
   }, [columns, columnCharts, chartsVisible, JSON.stringify(visibleColumns)]);
 
+  // Initialize the mini histogram charts toggle button.
+  // On load `chartsVisible` is set to `undefined`, the button will be disabled.
+  // Once we know how many rows have been returned,
+  // we decide whether to show or hide the charts by default.
+  useEffect(() => {
+    if (chartsVisible === undefined && rowCount > 0) {
+      setChartsVisible(rowCount < COLUMN_CHART_DEFAULT_VISIBILITY_ROWS_THRESHOLED);
```

Comment:
Does this check work when using a saved search for a data frame analytics job? I created a job off a search which matches about 7000 docs, but the histograms are not being shown. With the same saved search in the Transforms wizard, the histograms are displayed.

Reply:
Tested and confirmed fix in https://github.com/elastic/kibana/pull/92021/commits/e9a021f42e96e7da8b488420f8d409f0e3d5681b